docs/porting-extensions.md

```diff
@@ -251,35 +251,38 @@ In this diagram, each thread spindle symbol represents a thread that is running
 code inside a native extension. The lock icon indicates whether the thread holds
 the GIL - only one thread can acquire the GIL at a time, indicated by the
 fastened lock on the thread calling into the CPython C API. The bottom row of
-symbols incdicates what work each thread is doing. You can see that even with
+symbols indicates what work each thread is doing. You can see that even with
 the GIL it is possible to get multithreaded parallelism, so long as there are
 threads that do not have the GIL acquired and are not waiting to acquire the
 GIL. Usually, this means a thread is doing I/O or a long-running calculation
 that does not need any state or functionality from the CPython runtime.
 
 In addition to the lock icon indicating whether the GIL is acquired, each thread
-is either "plugged in" or "unplugged" from the interpreter runtime. This
-indicates whether a thread is "attached" to the runtime.
+icon is either plugged in or unplugged from the interpreter runtime, indicating
+it has an attached or detached thread state. In the GIL-enabled build, an
+attached thread holds the GIL and a detached thread cannot hold the GIL.
```

Comment:
This might be a bit clearer?

```suggestion
it has an attached or detached thread state. In the GIL-enabled build, only
one thread can have an attached thread state and hold the GIL, while all
other unplugged threads wait for the GIL to be released.
```